_docs/_latest/develop/dev/gcp-api.md

```diff
@@ -0,0 +1,213 @@
+---
+title: GCP API
+order: 103
+---
+
+# {{ page.title }}
+
+This page describes how to add a new GCP API client, which will allow
+Forseti to interact with GCP.
+
+---
+
+Before writing a new API client, it's important to note that GCP APIs have
+common methods:
+
+* list
+* get
+* create
+* update
+* delete
+
+Instead of writing these methods separately and repeatedly for all the
+different API clients, they can be written once as mixin classes, and then
+inherited by each API client. This makes it easy and quick to add new API
+clients, without having to rewrite these common methods for every API clients.
+
+You can [learn more about the mixin pattern here](https://www.ianlewis.org/en/mixins-and-python).
+
+You can see the currently available [mixins in Forseti here](https://github.com/GoogleCloudPlatform/forseti-security/blob/stable/google/cloud/forseti/common/gcp_api/repository_mixins.py).
+
+## Main Steps
+
+The main steps to add a new foo API client are:
+
+1. Define the API name & versions to be added
+1. Create a new foo API client file
+1. Create a FooClient class to provide the entry point to the GCP API method
+1. Create a FooRepositoryClient class, and define the property method for the resource that you want to interact with the GCP API
```

Comment:
I think it would be easier to understand if you also mention something like
FooClient contains FooRepositoryClient
FooRepositoryClient contains _FooBarRepository (as property)